Python-Functions-Decorators.md

```diff
@@ -1,9 +1,255 @@
 # Python Decorators
 
-[PEP 0318 - Decorators for Functions and Methods](https://www.python.org/dev/peps/pep-0318/)
 
-[Python Docs - Decorators](https://docs.python.org/3/whatsnew/2.4.html?highlight=decorator)
+Decorator essentially work as wrappers. They modify the behviour of the code before and after a target function execution, without the need to modify the function itself, augmenting the original functionality, thus decorating it.
 
-Now that we have covered basics of Python, let us apply Python in [Web Development](Web-Development-in-Python)!
+Before going in detail about decorators there are some concepts that should be clear. In Python, functions are objects and we can do a lot of useful stuff with them.
```

Comment:
> In Python, functions are objects

Link to some part of Python documentation that clearly states that, because a lot of readers who are just starting with Python would have trouble believing this.